code/procs/scorestats.dm

```diff
@@ -190,6 +200,93 @@ var/datum/score_tracker/score_tracker
 
 		return
 
+	/////////////////////////////////////
+	/////////Escapee stuff///////////////
+	/////////////////////////////////////
+	proc/calculate_escape_stats()
+		//set the global total to zero on proc start, just in cases.
+		richest_total = 0
+		//search mobs in centcom
+		for (var/mob/M in mobs)
+			var/area/A = get_area(M)
+			if (istype(A, /area/shuttle/escape/centcom))// || istype(A, /area/centcom))
+				if (!most_damaged_escapee)
+					most_damaged_escapee = M
+				else if (M.get_damage() < most_damaged_escapee.get_damage())
+					most_damaged_escapee = M
+
+				var/cash_total = get_cash_in_thing(M)
+				if (richest_total < cash_total)
+					richest_total = cash_total
+					richest_escapee = M
+
+		pets_escaped = list()
+
+		//This is shit, I know, but what else can I do? Take solace in the fact that this only ever runs once.
+		var/list/escape_contents = get_area_all_atoms(map_settings.escape_centcom)
+		for (var/obj/critter/C in escape_contents)
+			//Dr. Acula
+			if (istype(C, /obj/critter/bat/doctor))
+				var/obj/critter/bat/doctor/P = C
+				if (P.alive)
+					pets_escaped += P
+					acula_blood = P.blood_volume
+
+			//Sylvester
+			else if (istype(C, /obj/critter/turtle/sylvester))
+				if (C.alive)
+					pets_escaped += C
+
+			//Jones
+			else if (istype(C, /obj/critter/cat/jones))
+				if (C.alive)
+					pets_escaped += C
+
+			//George
+			else if (istype(C, /obj/critter/dog/george))
+				if (C.alive)
+					pets_escaped += C
+
+			//Heisenbee
+			else if (istype(C, /obj/critter/domestic_bee/heisenbee))
+				if (C.alive)
+					pets_escaped += C
+			//Don't really need to check for the mob variants. But maybe if we switch eventually??
+
+		if (locate(/obj/machinery/bot/secbot/beepsky) in world)
+			beepsky_alive = 1
+
+		return
+
+	proc/get_cash_in_thing(var/atom/A)
+		. = 0
+		for (var/I in A)
+			if (istype(I, /obj/item/storage))
+				. += get_cash_in_thing(I)
+			if (istype(I, /obj/item/spacecash))
+				var/obj/item/spacecash/SC = I
+				. += SC.amount
+			if (istype(I, /obj/item/card/id))
+				var/obj/item/card/id/ID = I
+				. += ID.amount
+		boutput(world, .)
+
+	proc/escapee_facts()
+		. = ""
+		//Richest Escapee | Most Damaged Escapee | Dr. Acula Blood Total | Clown Beatings
+		if (richest_escapee)		. += "<B>Richest Escapee:</B> [richest_escapee.real_name] : $[richest_total]<BR>"
+		if (most_damaged_escapee) 	. += "<B>Most Damaged Escapee:</B> [most_damaged_escapee.real_name] : [most_damaged_escapee.get_damage()]%<BR>"		//it'll be kinda different from when it's calculated, but whatever.
+		if (islist(pets_escaped))
+			var/who_escaped = ""
+			for (var/atom/A in pets_escaped)
+				who_escaped += " [A.name] [bicon(A)],"
+			. += "<B>Pets Escaped:</B>[copytext(who_escaped, 1, length(who_escaped)-1)]<BR><BR>"
```

Comment:
What's with this weird copytext?

Reply:
Uh, that's to remove the hanging comma at the end.

Reply:
It's a minor optimization in the grand scheme of things but probably use `var/list/who_escaped` and then do `who_escaped.Join(",")`, not only will it be faster but you'll also avoid the copytext.